README.md

```diff
@@ -423,6 +423,82 @@ Which will launch [torchrun](https://pytorch.org/docs/stable/elastic/run.html) w
 
 Alternatively, you can always subclass the core trainer and do things you way. There are many examples in the source dir and the end-to-end tests.
 
+## Streaming responses
+
+As a byte language model, MBLM generates integer representations of bytes. We can hook into the generation process and stream all generated bytes directly to a [file object](https://docs.python.org/3/glossary.html#term-file-object) such as `sys.stdout` (for debugging or interactive sessions) or `io.TextIO` and `io.BinaryIO` stream interfaces.
+
+Let's assume our model is conditioned to generate the following text string:
+
+```
+👉🏽 bytes generated by a 🤖
+```
+
+In UTF-8 bytes, this corresponds to:
+
+```sh
+# hex representation
+f0 9f 91 89 f0 9f 8f bd 20 62 79 74 65 73 20 67 65 6e 65 72 61 74 65 64 20 62 79 20 61 20 f0 9f a4 96
+
+# integer representation
+240 159 145 137 240 159 143 189 32 98 121 116 101 115 32 103 101 110 101 114 97 116 101 100 32 98 121 32 97 32 240 159 164 150
+```
+
+Internally, these integers are what the model generates. However, maybe you have trained to output a different modality such as a PNG file or an MP4 video - the possibilities are endless.
+
+For simplicity, let's assume we have some `root_dir` and a function `create_mblm` to create an MBLM module.
+
+### Streaming to a file
+
+We can **stream the response directly to a file** - no need to specify the encoding. All we need to do is open a file in binary mode. In this example, the output corresponds to UTF-8.
+
+```py
+from pathlib import Path
+
+from mblm.utils.stream import ByteStreamer
+
+mblm = create_mblm(...)
```

Comment:
Do we really need the mock with `create_mblm`? If the streamer is robust to partial corruption of UTF-8, even a random model should output some gibberish. Why not sth like
```py
mblm = MBLM(MBLMModelConfig(...
```

Reply:
That is correct. Using the ellipsis as a no-op placeholder simply (1) makes the snippet more concise and focuses on the generation instead of model construction and (2) lets us use actual assertions because we fake the "generated" bytes with the MBLM mock.

But I agree we should have a dedicated test for ensuring this also works with an actual MBLM. Added now.